WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/AdaptiveSheet.swift

```diff
@@ -52,7 +52,9 @@ struct AdaptiveSheetViewControllerRepresentable<Content: View>: UIViewController
     func updateUIViewController(_ uiViewController: AdaptiveSheetViewController<Content>, context: Context) {
         if isPresented {
             uiViewController.presentModalView()
-        } else if uiViewController.presentedViewController != nil {
+        } else if uiViewController.hostingViewController != nil,
+                  uiViewController.presentedViewController === uiViewController.hostingViewController {
+            // Dismiss if the view is presented
```

Comment:
2 questions here:
- Do we need the first nil check? Since we're confirming is not nil in the second check
- Do we need to check for identity with `===`, or checking equality with`==` would be enough?

```suggestion
        } else if  uiViewController.presentedViewController == uiViewController.hostingViewController {
            // Dismiss if the view is presented
```

Reply:
Good questions both of them:

> Do we need the first nil check? Since we're confirming is not nil in the second check

We don't confirm they are nil in the second check, if both are nil it returns true and dismisses the view controller: we don't want that.

> Do we need to check for identity with ===, or checking equality with== would be enough?

As here I want to compare if they are identically the same object, === makes it more explicit. == would work (Objective C == falls back to ===) but === makes it more clear in my opinion.